internal/controllers/inventory_controller.go

```diff
@@ -992,6 +1067,84 @@ func (t *reconcilerTask) createResourceServerClusterRoleBinding(ctx context.Cont
 	return nil
 }
 
+func (t *reconcilerTask) createClusterServerClusterRole(ctx context.Context) error {
+	role := &rbacv1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: fmt.Sprintf(
+				"%s-%s", t.object.Namespace, utils.InventoryClusterServerName,
+			),
+		},
+		Rules: []rbacv1.PolicyRule{
+			// We need to read managed clusters, as that is the main source for the
+			// information about clusters.
+			{
+				APIGroups: []string{
+					"cluster.open-cluster-management.io",
+				},
+				Resources: []string{
+					"managedclusters",
+				},
+				Verbs: []string{
+					"get",
+					"list",
+					"watch",
+				},
+			},
+			{
+				APIGroups: []string{
+					"agent-install.openshift.io",
+				},
+				Resources: []string{
+					"agents",
+				},
+				Verbs: []string{
+					"get",
+					"list",
+					"watch",
+				},
+			},
+		},
+	}
+
+	if err := utils.CreateK8sCR(ctx, t.client, role, t.object, utils.UPDATE); err != nil {
+		return fmt.Errorf("failed to create Cluster Server cluster role: %w", err)
+	}
+
+	return nil
+}
+
+func (t *reconcilerTask) createClusterServerClusterRoleBinding(ctx context.Context) error {
+	binding := &rbacv1.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: fmt.Sprintf(
+				"%s-%s",
+				t.object.Namespace, utils.InventoryClusterServerName,
+			),
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: rbacv1.GroupName,
+			Kind:     "ClusterRole",
+			Name: fmt.Sprintf(
+				"%s-%s",
+				t.object.Namespace, utils.InventoryClusterServerName,
+			),
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      rbacv1.ServiceAccountKind,
+				Namespace: t.object.Namespace,
+				Name:      utils.InventoryClusterServerName,
+			},
+		},
+	}
+
+	if err := utils.CreateK8sCR(ctx, t.client, binding, t.object, utils.UPDATE); err != nil {
+		return fmt.Errorf("failed to create Cluster Server cluster role binding: %w", err)
+	}
+
+	return nil
+}
+
 func (t *reconcilerTask) createAlarmServerClusterRole(ctx context.Context) error {
```

Comment:
could you please add rules here to fetch objects from the new endpoint. We only have for ResourceTypes

Reply:
Fixed